src/prefect/utilities/processutils.py

```diff
@@ -337,3 +337,4 @@ def kill_process(*args):
             os.kill(pid, signal.SIGKILL)
 
     signal.signal(signal.SIGINT, stop_process)
+    signal.signal(signal.SIGTERM, stop_process)
```

Comment:
We'll want a new utility called `forward_signal(signum: int, pid: int)` that'd look roughly like

```pyhton
You'll want to move this to the handler for the `prefect orion start` command and do something like

```python
previous_signal_handler = ...
def forward_signal
   os.kill(pid, signum)
   previous_signal_handler(..,)
   
signal.signal(sugnum, forward_signal)
```

Reply:
To test the change to the CLI, you can run `prefect orion start` in a subprocess (using `popen` or  similar) then send a SIGTERM to it. It should then exit gracefully.

Reply:
testing sounds good.

as for your utility suggestion, i think im stumbling over something. properly insufficient understanding of signals :)

do you mean a generalisation of the utility? sigterm calls signint calls sigkill? so recursion?

i think i dont understand `previous_signal_handler` 🤔

Reply:
is this something you'd like to implement, and could tag me on so i can learn?

Reply:
and by testing "exit gracefully", do you mean to poll the subprocess exit code and assert it is 1 within a second or two?

Reply:
do i understand correctly from the  [comment](https://github.com/PrefectHQ/prefect/blob/2.7.4/src/prefect/cli/orion.py#L141) in the code that you tried [builtin anyio signal handling](https://anyio.readthedocs.io/en/stable/signals.html), but it didn't work nicely together with click?

Reply:
Hi 👋 I took the time to put together the code and tests as I think you intended. The tests explicitly waits for `prefect orion start` to spin up a uvicorn worker, send two signals shortly after each other, and assert by the process output that the correct signals were received:
```
[
    b'\n',
    b' ___ ___ ___ ___ ___ ___ _____    ___  ___ ___ ___  _  _\n',
    b'| _ \\ _ \\ __| __| __/ __|_   _|  / _ \\| _ \\_ _/ _ \\| \\| |\n',
    b'|  _/   / _|| _|| _| (__  | |   | (_) |   /| | (_) | .` |\n',
    b'|_| |_|_\\___|_| |___\\___| |_|    \\___/|_|_\\___\\___/|_|\\_|\n',
    b'\n',
    b'Configure Prefect to communicate with the server with:\n',
    b'\n',
    b'    prefect config set PREFECT_API_URL=http://0.0.0.0:4200/api\n',
    b'\n',
    b'View the API reference documentation at http://0.0.0.0:4200/docs\n',
    b'\n',
    b"The dashboard is not built. It looks like you're on a development version.\n",
    b'See `prefect dev` for development commands.\n',
    b'\n',
    b'\n',
    b'\n',
    b'11:15:44.690 | INFO    | uvicorn.error - Started server process [89342]\n',
    b'11:15:44.693 | INFO    | uvicorn.error - Waiting for application startup.\n',
    b'11:15:44.993 | INFO    | prefect.orion - Scheduler service scheduled to start in-app\n',
    b'11:15:44.993 | INFO    | prefect.orion - RecentDeploymentsScheduler service scheduled to start in-app\n',
    b'11:15:44.994 | INFO    | prefect.orion - MarkLateRuns service scheduled to start in-app\n',
    b'11:15:44.994 | INFO    | prefect.orion - FailExpiredPauses service scheduled to start in-app\n',
    b'11:15:44.995 | INFO    | prefect.orion - Telemetry service scheduled to start in-app\n',
    b'11:15:44.995 | INFO    | prefect.orion - FlowRunNotifications service scheduled to start in-app\n',
    b'11:15:45.013 | INFO    | uvicorn.error - Application startup complete.\n',
    b'11:15:45.017 | INFO    | uvicorn.error - Uvicorn running on http://0.0.0.0:4200 (Press CTRL+C to quit)\n',
    b'\n',
    b'Sending SIGTERM to Orion (PID 89342)...\n',
    b'11:15:45.107 | INFO    | prefect.orion.services.recentdeploymentsscheduler - Scheduled 0 runs.\n',
    b'\n',
    b'Sending SIGKILL to Orion (PID 89342)...\n',
    b'Orion stopped!\n'
]
```

Reply:
added another test for the graceful shutdown. it looks _really_ graceful 😄 
```
 ___ ___ ___ ___ ___ ___ _____    ___  ___ ___ ___  _  _
| _ \ _ \ __| __| __/ __|_   _|  / _ \| _ \_ _/ _ \| \| |
|  _/   / _|| _|| _| (__  | |   | (_) |   /| | (_) | .` |
|_| |_|_\___|_| |___\___| |_|    \___/|_|_\___\___/|_|\_|

Configure Prefect to communicate with the server with:

    prefect config set PREFECT_API_URL=http://0.0.0.0:4200/api

View the API reference documentation at http://0.0.0.0:4200/docs

The dashboard is not built. It looks like you're on a development version.
See `prefect dev` for development commands.



11:38:26.690 | INFO    | uvicorn.error - Started server process [90293]
11:38:26.693 | INFO    | uvicorn.error - Waiting for application startup.
11:38:27.006 | INFO    | prefect.orion - Scheduler service scheduled to start in-app
11:38:27.007 | INFO    | prefect.orion - RecentDeploymentsScheduler service scheduled to start in-app
11:38:27.007 | INFO    | prefect.orion - MarkLateRuns service scheduled to start in-app
11:38:27.008 | INFO    | prefect.orion - FailExpiredPauses service scheduled to start in-app
11:38:27.008 | INFO    | prefect.orion - Telemetry service scheduled to start in-app
11:38:27.009 | INFO    | prefect.orion - FlowRunNotifications service scheduled to start in-app
11:38:27.029 | INFO    | uvicorn.error - Application startup complete.
11:38:27.031 | INFO    | uvicorn.error - Uvicorn running on http://0.0.0.0:4200 (Press CTRL+C to quit)

Sending SIGTERM to Orion (PID 90293)...
11:38:27.147 | INFO    | uvicorn.error - Shutting down
11:38:27.152 | INFO    | prefect.orion.services.recentdeploymentsscheduler - Scheduled 0 runs.
11:38:27.155 | INFO    | prefect.orion.services.scheduler - Scheduled 0 runs.
11:38:27.156 | INFO    | prefect.orion.services.failexpiredpauses - Finished monitoring for late runs.
11:38:27.158 | INFO    | prefect.orion.services.marklateruns - Finished monitoring for late runs.
11:38:27.161 | INFO    | prefect.orion - FlowRunNotifications service stopped!
11:38:27.250 | INFO    | uvicorn.error - Waiting for application shutdown.
11:38:27.781 | INFO    | prefect.orion - Telemetry service stopped!
11:38:28.159 | INFO    | prefect.orion - RecentDeploymentsScheduler service stopped!
11:38:28.160 | INFO    | prefect.orion - Scheduler service stopped!
11:38:28.161 | INFO    | prefect.orion - FailExpiredPauses service stopped!
11:38:28.162 | INFO    | prefect.orion - MarkLateRuns service stopped!
11:38:28.175 | INFO    | uvicorn.error - Application shutdown complete.
11:38:28.176 | INFO    | uvicorn.error - Finished server process [90293]
Orion stopped!
```